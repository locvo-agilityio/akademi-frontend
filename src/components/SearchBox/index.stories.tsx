import type { Meta, StoryObj } from '@storybook/react';

// Components
import SearchBox from '.';

const meta = {
  title: 'Components/SearchBox',
  component: SearchBox,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
} satisfies Meta<typeof SearchBox>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    onChange: (value: string) => console.log(value),
  },
};
