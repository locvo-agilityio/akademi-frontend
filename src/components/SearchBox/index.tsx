'use client';

import { ChangeEvent, memo, useCallback } from 'react';
import {
  Icon,
  Input,
  InputGroup,
  InputLeftElement,
  InputProps,
} from '@chakra-ui/react';

// Icons
import { SearchIcon } from '@/components/icons';

type TSearchBoxProps = Omit<InputProps, 'onChange'> & {
  onChange: (value: string) => void;
};

const SearchBox = ({ onChange, ...rest }: TSearchBoxProps) => {
  const handleChangeValue = useCallback(
    (e: ChangeEvent<HTMLInputElement>): void => onChange(e.target.value),
    [onChange],
  );

  return (
    <InputGroup maxW={350}>
      <Input
        type="text"
        variant="secondary"
        placeholder="Search here..."
        onChange={handleChangeValue}
        {...rest}
      />
      <InputLeftElement
        pointerEvents="none"
        children={<Icon as={SearchIcon} boxSize={5} />}
        top={2}
        left={2}
      />
    </InputGroup>
  );
};

export default memo(SearchBox);
