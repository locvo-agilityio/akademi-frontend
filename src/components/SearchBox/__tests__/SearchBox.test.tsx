import { fireEvent, render } from '@testing-library/react';

// Components
import SearchBox from '..';

const mockOnChange = jest.fn();

describe('SearchBox Component', () => {
  it('Match SearchBox component', () => {
    const element = render(<SearchBox onChange={mockOnChange} />);

    expect(element).toMatchSnapshot();
  });

  it('Should call onChange function', () => {
    const { getByPlaceholderText } = render(
      <SearchBox onChange={mockOnChange} />,
    );

    const input = getByPlaceholderText('Search here...') as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'Hello' } });

    expect(mockOnChange).toHaveBeenCalled();
  });
});
