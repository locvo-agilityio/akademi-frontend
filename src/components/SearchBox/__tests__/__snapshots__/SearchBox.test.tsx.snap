// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component Match SearchBox component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="chakra-input__group css-29d2s5"
        data-group="true"
      >
        <input
          class="chakra-input css-0"
          placeholder="Search here..."
          type="text"
        />
        <div
          class="chakra-input__left-element css-d9lws8"
        >
          <svg
            class="chakra-icon css-1vp16hi"
            fill="none"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M23.6 21.8001L18 16.2001C19.3 14.5001 20.1 12.4001 20.1 10.1001C20.1 4.6001 15.6 0.100098 10.1 0.100098C4.6 0.100098 0 4.6001 0 10.1001C0 15.6001 4.5 20.1001 10 20.1001C12.3 20.1001 14.5 19.3001 16.2 17.9001L21.8 23.5001C22 23.7001 22.4 23.9001 22.7 23.9001C23 23.9001 23.3 23.8001 23.6 23.5001C24.1 23.1001 24.1 22.3001 23.6 21.8001ZM2.5 10.1001C2.5 6.0001 5.9 2.6001 10 2.6001C14.1 2.6001 17.5 6.0001 17.5 10.1001C17.5 14.2001 14.1 17.6001 10 17.6001C5.9 17.6001 2.5 14.3001 2.5 10.1001Z"
              fill="#4D44B5"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-input__group css-29d2s5"
      data-group="true"
    >
      <input
        class="chakra-input css-0"
        placeholder="Search here..."
        type="text"
      />
      <div
        class="chakra-input__left-element css-d9lws8"
      >
        <svg
          class="chakra-icon css-1vp16hi"
          fill="none"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M23.6 21.8001L18 16.2001C19.3 14.5001 20.1 12.4001 20.1 10.1001C20.1 4.6001 15.6 0.100098 10.1 0.100098C4.6 0.100098 0 4.6001 0 10.1001C0 15.6001 4.5 20.1001 10 20.1001C12.3 20.1001 14.5 19.3001 16.2 17.9001L21.8 23.5001C22 23.7001 22.4 23.9001 22.7 23.9001C23 23.9001 23.3 23.8001 23.6 23.5001C24.1 23.1001 24.1 22.3001 23.6 21.8001ZM2.5 10.1001C2.5 6.0001 5.9 2.6001 10 2.6001C14.1 2.6001 17.5 6.0001 17.5 10.1001C17.5 14.2001 14.1 17.6001 10 17.6001C5.9 17.6001 2.5 14.3001 2.5 10.1001Z"
            fill="#4D44B5"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
